src/main/scala/nl.knaw.dans.easy.file2bag/FilesXml.scala

```diff
@@ -31,27 +31,41 @@ object FilesXml extends DebugEnhancedLogging {
     // easy-deposit-api does not supply rights at all
     // so here we don't supply a default
     // https://github.com/DANS-KNAW/easy-deposit-api/blob/eef71618e8b776fb274da123f8011510499c741a/src/main/scala/nl.knaw.dans.easy.deposit/docs/FilesXml.scala#L40-L42
-    val itemContent = if (accessRights.isBlank) Seq[Node]()
-                      else <accessibleToRights>{ accessRights }</accessibleToRights>
-    val newItem =
-        <file filepath={ "data/" + destinationPath }>
-          <dcterms:format>{ mimeType }</dcterms:format>
-          { itemContent }
-        </file>
-    logger.info(newItem.toOneLiner)
+    var formatTagPrefix = "dcterms"
+    val oldFilesXmlWithPossiblyAddedNamespace =
+      if ((oldFilesXml \ "file" \ "format").nonEmpty) {
+        formatTagPrefix = (oldFilesXml \ "file" \ "format").head.prefix
+        oldFilesXml
+      }
+      else {
+        if (oldFilesXml.scope.getURI(formatTagPrefix) == null) {
+          oldFilesXml.copy(scope = NamespaceBinding(formatTagPrefix, "http://purl.org/dc/terms/", oldFilesXml.scope))
+        }
+        else
+          oldFilesXml
+      }
+    val accessibleTo = if (accessRights.isBlank) ""
+                       else s"<accessibleToRights>$accessRights</accessibleToRights>"
+    val newFileElement =
+      XML.loadString(
+        s"""<file filepath="data/$destinationPath">
+           <$formatTagPrefix:format>$mimeType</$formatTagPrefix:format>
+           $accessibleTo
+        </file>""")
+    logger.info(newFileElement.toOneLiner)
 
     object insertElement extends RewriteRule {
       override def transform(node: Node): Seq[Node] = node match {
         case Elem(boundPrefix, "files", _, boundScope, children @ _*) =>
           <files>
             { children }
-            { newItem }
+            { newFileElement }
```

Comment:
what if there is no `files.xml` file at all?

Reply:
if there is no files.xml this method is not called at all

Reply:
what i meant to ask: if there is no `files.xml`, will such a file be created? Or will the files be added to the bag, but not be reflected in the files.xml as it does not exist? In that case, the bag will not be valid.